src/core/syscalls/os_syscall_handler.rs

```diff
@@ -1098,4 +847,45 @@ mod tests {
             })
         )
     }
+
+    #[test]
+    fn call_contract_and_write_response_err() {
+        let retdata_iterator = vec![vec![12].into(), vec![6543].into()].into();
+        let mut handler = OsSyscallHandler {
+            retdata_iterator,
+            ..Default::default()
+        };
+
+        let mut vm = vm!();
+        let syscall_ptr = Relocatable {
+            segment_index: 0,
+            offset: 0,
+        };
+        assert_eq!(
+            handler._call_contract_and_write_response("call_contract", &mut vm, syscall_ptr),
+            Err(SyscallHandlerError::VirtualMachineError(
+                VirtualMachineError::MemoryError(MemoryError::UnallocatedSegment(0, 0))
+            ))
+        )
+    }
+    #[test]
+    fn call_contract_and_write_response() {
+        let retdata_iterator = vec![vec![12].into(), vec![6543].into()].into();
+        let mut handler = OsSyscallHandler {
+            retdata_iterator,
+            ..Default::default()
+        };
+
+        let mut vm = vm!();
+        let syscall_ptr = Relocatable {
+            segment_index: 0,
+            offset: 0,
+        };
+
+        vm.add_memory_segment();
+        assert_eq!(
+            handler._call_contract_and_write_response("call_contract", &mut vm, syscall_ptr),
+            Ok(())
```

Comment:
Check the reponse insert `self._write_syscall_response(&response, vm, syscall_ptr)`

Reply:
Just add this assert and we will be good to go.